frontend/src/customizations/volto/components/theme/Footer/Footer.jsx

```diff
@@ -99,6 +99,11 @@ const Footer = ({ intl }) => {
                   <List.Item>
                     <a href="https://ploneconf.org">Conference</a>
                   </List.Item>
+                  <List.Item>
+                    <a href="https://plone.org/news-and-events/join-the-plone-newsletter">
```

Comment:
The link is relative to the site root.

```suggestion
                    <a href="/news-and-events/join-the-plone-newsletter">
```